lib/virtual-net/src/host.rs

```diff
@@ -103,16 +103,32 @@ impl VirtualNetworking for LocalNetworking {
     async fn bind_udp(
         &self,
         addr: SocketAddr,
-        _reuse_port: bool,
-        _reuse_addr: bool,
+        reuse_port: bool,
+        reuse_addr: bool,
     ) -> Result<Box<dyn VirtualUdpSocket + Sync>> {
+        use socket2::{Domain, Socket, Type};
+
         if let Some(ruleset) = self.ruleset.as_ref() {
             if !ruleset.allows_socket(addr, Direction::Inbound) {
                 tracing::warn!(%addr, "bind_udp blocked by firewall rule");
                 return Err(NetworkError::PermissionDenied);
             }
         }
 
+        #[cfg(not(windows))]
```

Comment:
Any reason why we are not using the same logic for Windows? It seems that the API for socket reusing is also available there, so the more commonalities between systems, the better